relay-general/src/pii/minidumps.rs

```diff
@@ -130,21 +240,64 @@ impl PiiAttachmentsProcessor<'_> {
         data: &mut [u8],
     ) -> Result<bool, ScrubMinidumpError> {
         let file_state = self.state(filename, ValueType::Minidump);
-        let streams = MinidumpData::parse(data)?.streams()?;
+        let items = MinidumpData::parse(data)?.items()?;
         let mut changed = false;
 
-        for StreamDescriptor(stream_type, value_type, range) in streams {
-            let slice = data
-                .get_mut(range)
-                .ok_or(ScrubMinidumpError::InvalidAddress(stream_type))?;
-
+        for item in items {
             // IMPORTANT: Minidump sections are always classified as Pii:Maybe. This avoids to
             // accidentally scrub stack memory with highly generic selectors. TODO: Update the PII
             // system with a better approach.
             let attrs = Cow::Owned(FieldAttrs::new().pii(Pii::Maybe));
 
-            let state = file_state.enter_static("", Some(attrs), Some(value_type));
-            changed |= self.scrub_bytes(slice, &state);
+            match item {
+                MinidumpItem::StackMemory(range) => {
+                    let slice = data
+                        .get_mut(range)
+                        .ok_or(ScrubMinidumpError::InvalidAddress)?;
+                    let state =
+                        file_state.enter_static("", Some(attrs), Some(ValueType::StackMemory));
+                    changed |= self.scrub_bytes(slice, &state, ScrubEncodings::All);
+                }
+                MinidumpItem::NonStackMemory(range) => {
+                    let slice = data
+                        .get_mut(range)
+                        .ok_or(ScrubMinidumpError::InvalidAddress)?;
+                    let state =
+                        file_state.enter_static("", Some(attrs), Some(ValueType::HeapMemory));
+                    changed |= self.scrub_bytes(slice, &state, ScrubEncodings::All);
+                }
+                MinidumpItem::LinuxEnviron(range) | MinidumpItem::LinuxCmdLine(range) => {
+                    let slice = data
+                        .get_mut(range)
+                        .ok_or(ScrubMinidumpError::InvalidAddress)?;
+                    let state = file_state.enter_static("", Some(attrs), Some(ValueType::Binary));
+                    changed |= self.scrub_bytes(slice, &state, ScrubEncodings::All);
+                }
+                MinidumpItem::CodeModuleName(range) => {
+                    let slice = data
+                        .get_mut(range)
+                        .ok_or(ScrubMinidumpError::InvalidAddress)?;
+                    let state = file_state.enter_static(
+                        "code_file",
+                        Some(attrs),
+                        Some(ValueType::Filepath),
+                    );
+                    let wstr = WStr::from_utf16le_mut(slice)?; // TODO: Consider making this lossy?
+                    changed |= self.scrub_utf16_filepath(wstr, &state);
```

Comment:
```
"applications": {
    "code_file": ["anything:remove"]
}
``` 
currently still only gives me things like:
```
(code_file)                     = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\crash.exe"
```
because `scrub_utf16_filepath` never tries to scrub the basename. Is that what we want?

Reply:
i was expecting that for `"$filepath": ["anything:remove"]` but I'm probably misunderstanding selectors and ValueTypes

Reply:
This lgtm. `$filepath` should behave the same (it's just an alias)

Reply:
as i understand it it makes it impossible to completely remove these names as it will always be identified as this valuetype and only do partial scrubing.  as long as we want that, that's fine by me.

Reply:
Yes we want that specifically because we don't want people to be able to break stacktrace processing with data scrubbing. Stacktrace processing needs the basename.